apps/api/src/domain/models/digital-text/commands/create-digital-text.command.integration.spec.ts

```diff
@@ -0,0 +1,227 @@
+import { generateCommandFuzzTestCases } from '../../__tests__/command-helpers/generate-command-fuzz-test-cases';
+import { CommandAssertionDependencies } from '../../__tests__/command-helpers/types/CommandAssertionDependencies';
+import buildDummyUuid from '../../__tests__/utilities/buildDummyUuid';
+import { dummySystemUserId } from '../../__tests__/utilities/dummySystemUserId';
+import { dummyUuid } from '../../__tests__/utilities/dummyUuid';
+import CommandExecutionError from '../../shared/common-command-errors/CommandExecutionError';
+import UuidNotGeneratedInternallyError from '../../shared/common-command-errors/UuidNotGeneratedInternallyError';
+import { DigitalText } from '../digital-text.entity';
+import { CreateDigitalText } from './create-digital-text.command';
+
+const createDigitalTextCommandType = 'CREATE_DIGITAL_TEXT';
+
+const buildValidCommandFSA = (id: AggregateId): FluxStandardAction<DTO<CreateDigitalText>> => ({
+    type: createDigitalTextCommandType,
+    payload: {
+        aggregateCompositeIdentifier: { id, type: AggregateType.digitalText },
+        title: 'test-digital-text-name (language)',
+        languageCodeForTitle: LanguageCode.Haida,
+    },
+});
+
+const buildInvalidFSA = (
+    id: AggregateId,
+    payloadOverrides: Partial<Record<keyof CreateDigitalText, unknown>> = {}
+): FluxStandardAction<DTO<CreateDigitalText>> => ({
+    type: createDigitalTextCommandType,
+    payload: {
+        ...buildValidCommandFSA(id).payload,
+        ...(payloadOverrides as Partial<CreateDigitalText>),
+    },
+});
+
+const initialState = buildInMemorySnapshot({});
+
+describe('CreateDigitalText', () => {
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let commandHandlerService: CommandHandlerService;
+
+    let app: INestApplication;
+
+    let databaseProvider: ArangoDatabaseProvider;
+
+    let idManager: IIdManager;
+
+    let assertionHelperDependencies: CommandAssertionDependencies;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, idManager, app, databaseProvider } =
+            await setUpIntegrationTest({
+                ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+            }));
+
+        assertionHelperDependencies = {
+            testRepositoryProvider,
+            commandHandlerService,
+            idManager,
+        };
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    afterEach(async () => {
+        await testRepositoryProvider.testTeardown();
+    });
+
+    afterAll(async () => {
+        await app.close();
+
+        databaseProvider.close();
+    });
+
+    describe('when the payload is valid', () => {
+        it('should succeed', async () => {
+            await assertCreateCommandSuccess(assertionHelperDependencies, {
+                buildValidCommandFSA,
+                initialState,
```

Comment:
Does `assertCreateCommandSuccess` have an alternate overload with `seedInitialState`? If not, I need to add it. The point is, we're moving towards seeding state with commands instead of snapshots, and so I generalized the API of the command helpers to have an async `seedInitialState` where you can run any logic you want. You're still allowed to `testRepositoryProvider.addFullSnapshot(...)` for now, but that will change soon. The reason is, when we event source, we need an event history to work with in order for the event-sourced command repositories to actually have the correct behaviour.

Reply:
Doesn't look like it exists yet

Reply:
Done.